adagrams/game.py

```diff
@@ -1,11 +1,145 @@
+
+import random
+
 def draw_letters():
-    pass
+    letter_pool = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+    letters = ""
+    for letter , num in letter_pool.items():
+        letters += letter * num
+    
+    result = []
+    letters_count_dict = {
+    'A': 0, 
+    'B': 0, 
+    'C': 0, 
+    'D': 0, 
+    'E': 0, 
+    'F': 0, 
+    'G': 0, 
+    'H': 0, 
+    'I': 0, 
+    'J': 0, 
+    'K': 0, 
+    'L': 0, 
+    'M': 0, 
+    'N': 0, 
+    'O': 0, 
+    'P': 0, 
+    'Q': 0, 
+    'R': 0, 
+    'S': 0, 
+    'T': 0, 
+    'U': 0, 
+    'V': 0, 
+    'W': 0, 
+    'X': 0, 
+    'Y': 0, 
+    'Z': 0
+    }
+    
+    while len(result) < 10:
+        i = random.randint(0, len(letters) - 1)
+        if letter_pool[letters[i]] > letters_count_dict[letters[i]]:
+            letters_count_dict[letters[i]] += 1
+            result.append(letters[i])
+    
+    return result
 
 def uses_available_letters(word, letter_bank):
-    pass
+    letter_bank_copy = []
+    word_lower = word.lower()
+    for letter in letter_bank:
+        letter_bank_copy.append(letter.lower())
```

Comment:
Instead of creating a variable to store the word with the case changed, you can directly call the function in the loop, like this:

```python
for letter in word.lower()
    # rest of your code
```

Reply:
There are a couple of other ways to copy a list:
```python
# use list comprehension
letter_bank_copy = [letter for letter in letter_bank]

# use string slicing
letter_bank_copy = letter_bank[:]
```